travis/shared_configs/python_tester.yml

```diff
@@ -0,0 +1,12 @@
+python:
+  - "3.6"
+  - "3.7"
+  - "3.8"
+
+install:
+  - pip install -r requirements.txt
+  - pip install -r dev-requirements.txt
+
+script:
+  - coverage run run_tests.py
```

Comment:
I have a local script I use that runs tests based on the presence/absence of `run_tests.py` called `rt`:
```bash
if command -v coverage; then
    if [ -f "run_tests.py" ]; then
        coverage run --concurrency=thread --parallel-mode run_tests.py "$@"
    else
        coverage run --concurrency=thread --parallel-mode -m pytest "$@"
    fi

    # (coverage combine && coverage report | grep -v -e ' 0%') || /usr/bin/true
    coverage combine || /usr/bin/true
else
    if [ -f "run_tests.py" ]; then
        python run_tests.py --concurrency=thread --parallel-mode "$@"
    else
        python -m pytest "$@"
    fi
fi
```

Probably should grab the useful bits of the above out and use them here. NSLS-II at least started removing their `run_tests.py` files as they aren't so useful.

Reply:
🔥